File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookParagraphDao.java
package ai.elimu.content_provider.room.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ai.elimu.content_provider.room.entity.StoryBookParagraph;

@Dao
public interface StoryBookParagraphDao {

    @Insert
    void insert(StoryBookParagraph storyBookParagraph);

    @Query("SELECT * FROM StoryBookParagraph WHERE id = :id")
    StoryBookParagraph load(Long id);

    @Query("SELECT * FROM StoryBookParagraph WHERE id = :id")
    Cursor loadAsCursor(Long id);

    @Query("SELECT * FROM StoryBookParagraph WHERE storyBookChapterId = :storyBookChapterId")
    List<StoryBookParagraph> loadAll(Long storyBookChapterId);

    @Query("SELECT * FROM StoryBookParagraph WHERE storyBookChapterId = :storyBookChapterId")
    Cursor loadAllAsCursor(Long storyBookChapterId);

    @Update
    void update(StoryBookParagraph storyBookParagraph);
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookParagraphDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.StoryBookParagraph
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface StoryBookParagraphDao {
    @Insert
    fun insert(storyBookParagraph: StoryBookParagraph)

    @Query("SELECT * FROM StoryBookParagraph WHERE id = :id")
    fun load(id: Long?): StoryBookParagraph?

    @Query("SELECT * FROM StoryBookParagraph WHERE id = :id")
    fun loadAsCursor(id: Long?): Cursor?

    @Query("SELECT * FROM StoryBookParagraph WHERE storyBookChapterId = :storyBookChapterId")
    fun loadAll(storyBookChapterId: Long?): MutableList<StoryBookParagraph?>?

    @Query("SELECT * FROM StoryBookParagraph WHERE storyBookChapterId = :storyBookChapterId")
    fun loadAllAsCursor(storyBookChapterId: Long?): Cursor

    @Update
    fun update(storyBookParagraph: StoryBookParagraph)

    @Query("DELETE FROM StoryBookParagraph")
    fun deleteAll()
}
